File: GFG/Queue/stonq.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

class mSt{
    public:
    mSt() : sz(0) {}
    Queue<int> q1,q2;
    int sz;
    int size(){
        return sz;
    }
    bool isEmpty(){
        return sz==0;
    }
    void push(int data){
        while(q1.isEmpty()==false){
            q2.enque(q1.deque());
        }
        q1.enque(data);
        while(q2.isEmpty()==false){
            q1.enque(q2.deque());
        }
        sz++;
    }
    int pop(){
        sz--;
        return q1.deque();
    }
    int peek(){
        return q1.front();
    }
};

int main(){
    mSt s;
    for(int i=1;i<11;i++){
        s.push(i*10);
    }
    cout<<s.size()<<" "<<s.isEmpty()<<endl;
    while(s.isEmpty()==false){
        cout<<s.pop()<<" ";
    }
    return 0;
}

File: GFG/Queue/stonq.py
import ds

class mSt:
    def __init__(self):
        self.q1=ds.Queue()
        self.q2=ds.Queue()
        self.sz=0
    def size(self):
        return self.sz
    def isEmpty(self):
        return self.sz==0
    def top(self):
        return self.q1.front()
    def push(self,data):
        while(self.q1.isEmpty()==False):
            self.q2.enqueue(self.q1.deque())
        self.q1.enqueue(data)
        while(self.q2.isEmpty()==False):
            self.q1.enqueue(self.q2.deque())
        self.sz+=1
    def pop(self):
        self.sz-=1
        return self.q1.deque()
    
st=mSt()
for i in range(1,11):
    st.push(i*10)
print(st.size()," ",st.isEmpty())
while(st.isEmpty()==False):
    print(st.pop(),end=" ")